src/services/similarity.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @async
+ * @param {number} recipient_id - The ID of the recipient.
+ * @param {boolean} [cluster] - Specifies whether to cluster the results. Default value is false.
+ * @returns {Promise<Array>} A promise that resolves to an array of similar goals.
+ */
+export default async function getSimilarGoalsForRecipient(recipient_id, cluster) {
+  const { SIMILARITY_ENDPOINT } = process.env;
+
+  const response = await fetch(
```

Comment:
should this fetch be within the try?

Reply:
it 100% should be: 0e7c21352b5c958802a655d8e929847e8a9fe9ff